src/scalarstats.jl

```diff
@@ -348,14 +397,147 @@ function summarystats{T<:Real}(a::AbstractArray{T})
         convert(R, qs[5]))
 end
 
+function summarystats{T, W<:Real}(a::AbstractVector{T}, w::WeightVec{W})
+    m = mean(a, w)
+    qs = quantile(a, [0.00, 0.25, 0.50, 0.75, 1.00], w)
+    R = typeof(convert(FloatingPoint, zero(T)))
+    WeightedSummaryStats{R, W}(
+        convert(R, m),
+        convert(R, qs[1]),
+        convert(R, qs[2]),
+        convert(R, qs[3]),
+        convert(R, qs[4]),
+        convert(R, qs[5]),
+        w.sum)
+end
+
+
+function detailedsummarystats{T<:Real}(a::AbstractArray{T})
+    m = mean(a)
+    std = stdm(a, m)
+    sk = skewness(a, m)
+    ku = kurtosis(a, m)
+    qs = quantile(a, [0.00, 0.01, 0.05, 0.10, 0.25, 0.50, 0.75, 0.90, 0.95, 0.99, 1.00])
+    R = typeof(convert(FloatingPoint, zero(T)))
+    DetailedSummaryStats{R}(
+        convert(R, m),
+        convert(R, std),
+        convert(R, sk),
+        convert(R, ku),
+        convert(R, qs[1]),
+        convert(R, qs[2]),
+        convert(R, qs[3]),
+        convert(R, qs[4]),
+        convert(R, qs[5]),
+        convert(R, qs[6]),
+        convert(R, qs[7]),
+        convert(R, qs[8]),
+        convert(R, qs[9]),
+        convert(R, qs[10]),
+        convert(R, qs[11]))
+end
+
+function detailedsummarystats{T, W<:Real}(a::AbstractVector{T}, w::WeightVec{W})
+    m = mean(a, w)
+    std = stdm(a, m, w)
+    sk = skewness(a, w, m)
+    ku = kurtosis(a, w, m)
+    qs = quantile(a, w, [0.00, 0.01, 0.05, 0.10, 0.25, 0.50, 0.75, 0.90, 0.95, 0.99, 1.00])
+    R = typeof(convert(FloatingPoint, zero(T)))
+    WeightedDetailedSummaryStats{R, W}(
+        convert(R, m),
+        convert(R, std),
+        convert(R, sk),
+        convert(R, ku),
+        convert(R, qs[1]),
+        convert(R, qs[2]),
+        convert(R, qs[3]),
+        convert(R, qs[4]),
+        convert(R, qs[5]),
+        convert(R, qs[6]),
+        convert(R, qs[7]),
+        convert(R, qs[8]),
+        convert(R, qs[9]),
+        convert(R, qs[10]),
+        convert(R, qs[11]),
+        w.sum)
+end
+
+
+
+
 function Base.show(io::IO, ss::SummaryStats)
     println(io, "Summary Stats:")
     @printf(io, "Mean:         %.6f\n", ss.mean)
     @printf(io, "Minimum:      %.6f\n", ss.min)
-    @printf(io, "1st Quartile: %.6f\n", ss.q25)
+    @printf(io, "1st Quartile: %.6f\n", ss.p25)
+    @printf(io, "Median:       %.6f\n", ss.median)
+    @printf(io, "3rd Quartile: %.6f\n", ss.p75)
+    @printf(io, "Maximum:      %.6f\n", ss.max)
+end
+
+function Base.show(io::IO, ss::WeightedSummaryStats)
```

Comment:
There's a lot of duplication here. Can't you call the unweighted version, and then output the sum of weights?